### PR TITLE
feat: refactor kadena-cli types and fix issues in commands

### PR DESCRIPTION
**Pull request by barthuijgen**

### CODE REVIEW

File: packages/tools/kadena-cli/src/keys/commands/keysDeleteWallet.ts [hunk 1]
```diff
@@ -31,18 +65,23 @@ export const createDeleteKeysCommand: (
       });
 
       if (config.keyWallet === 'all') {
-        const confirmDelete = await externalPrompt.confirmWalletDeletePrompt();
-        if (confirmDelete === 'no') {
-          console.log(chalk.yellow('\nNo wallets were deleted.\n'));
-          return;
+        if (config.quiet !== true) {
```
Comment (KdWarlordz) [inline packages/tools/kadena-cli/src/keys/commands/keysDeleteWallet.ts:68]: 👍

File: packages/tools/kadena-cli/src/utils/createOption.ts [hunk 2]
```diff
@@ -1,56 +1,91 @@
...
+      option,
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      prompt: (responses: any, args: any) =>
-        option.prompt(responses, args, isOptional),
-      validation: isOptional ? option.validation.optional() : option.validation,
+      expand: expand as T['expand'] extends (...args: any[]) => unknown
+        ? (value: Awaited<ReturnType<T['prompt']>>) => ReturnType<T['expand']>
+        : undefined,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      transform: transform as T['transform'] extends (...args: any[]) => unknown
```
Comment (KdWarlordz) [inline packages/tools/kadena-cli/src/utils/createOption.ts:58]: We should go through this.

**Review by KdWarlordz — APPROVED** [review]

File: packages/tools/kadena-cli/src/utils/createOption.ts (hunk 2 above)
Comment (barthuijgen) [reply packages/tools/kadena-cli/src/utils/createOption.ts:58]: Would be glad to. Send me a message on slack.

File: packages/tools/kadena-cli/src/utils/createOption.ts [hunk 3]
```diff
@@ -1,56 +1,91 @@
...
+      prompt: ((responses, args) =>
+        prompt(responses, args, optional)) as T['prompt'],
+      validation: (isOptional === true
+        ? validation.optional()
+        : validation) as T['validation'],
     };
   };
 }
+
+const testoption = createOption({
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/utils/createOption.ts:74]: `testOption` is test code, I will remove it after discussion about `createOption` is done.

**Review by alber70g — APPROVED** [review]

**Pull request merged**